File: src/pythonintro/Student.java
package pythonintro;

public class Student {

    private String _name;
    private int _idNumber;
    private String _concentration;
    private int _graduationYear;

    /* Student contructor which takes in a graduation year */
    public Student(String name, int idNumber, String concentration, int graduationYear) {
        _name = name;
        _idNumber = idNumber;
        _graduationYear = graduationYear;
        _concentration = concentration;
    }

    /* Student overloading! Second contructor which has a default graduation year of 2020, in case only three arguments are given. */
    public Student(String name, int idNumber, String concentration) {
        _name = name;
        _idNumber = idNumber;
        _graduationYear = 2020;
        _concentration = concentration;
    }

    public void setConcentration(String concentration) {
        _concentration = concentration;
    }

    public void printStudentInfo() {
//        System.out.println("Student named " + _name + " has ID number " + _idNumber.toString() + " is graduating in " _graduationYear.toString() + " and is studying " + _concentration + ".")
    }
}

File: cs16/pythonintro/student.py
class Student:
    """The student class models a student with a name, ID number,
    a graduation year, and a concentration.
    """
    def __init__(self, name, idNumber, concentration, graduationYear=2020):
        self._name = name
        self._idNumber = idNumber
        self._graduationYear = graduationYear
        self._concentration = concentration

    def set_concentration(self, concentration):
        self._concentration = concentration

    # Other accessors/mutators...
    def print_student_info(self):
        print("Student named " + self._name + " has ID number " + \
            str(self._idNumber) + ", is graduating in " + \
            str(self._graduationYear) + " and is studying " + \
            self._concentration + ".")

if __name__ == "__main__" :
    dara = Student("Dara", 1002354, "Physics")
    dara.set_concentration("Computer Science")
    dara.print_student_info()